cowprotocol/cowswap
# chore(recipient-warning): 3rd party recipient banner wire up

# Update 2023-09-15

- Added to swap confirm modal as well 
![image](https://github.com/cowprotocol/cowswap/assets/43217/580cf649-582b-4489-a00e-a1de2081bc0e)

- And limit
![image](https://github.com/cowprotocol/cowswap/assets/43217/63b24d70-dd53-45d5-9d79-2de972b25cbc)

- And TWAP
![image](https://github.com/cowprotocol/cowswap/assets/43217/e5f63d7b-0da5-484e-acdc-93519fa8f2e5)

- Updated text to be shorter and generic

## NOT included
- Styling
- Showing warning in the table row

# Summary

Wiring up banner logic and cleaning up the local storage in favor of a Jotai atom.

# To Test

1. Place SWAP sending to a different recipient
* Activity details should show the warning
<img width="829" alt="image" src="https://github.com/cowprotocol/cowswap/assets/43217/1f55020b-3506-4ada-a09e-a071f6d8e68e">
2. Click on dismiss
* Warning should be gone
3. Refresh the page and open activity details again
* Warning should not be shown

4. Place LIMIT order sending to a different recipient
* Order details should show the warning
<img width="497" alt="image" src="https://github.com/cowprotocol/cowswap/assets/43217/80fc2973-3850-4b73-b7de-27bcce6966cb">

5. Click on dismiss
* Warning should be gone
6. Refresh the page and open the order details again
* Warning should not be shown
